Sources/Testing/Discovery.swift

```diff
@@ -24,11 +44,27 @@ private import _TestingInternals
 public typealias __TestContentRecord = (
   kind: UInt32,
   reserved1: UInt32,
-  accessor: (@convention(c) (_ outValue: UnsafeMutableRawPointer, _ hint: UnsafeRawPointer?) -> CBool)?,
+  accessor: __TestContentRecordAccessor?,
   context: UInt,
   reserved2: UInt
 )
 
+/// Check that two Swift types are equivalent.
+///
+/// - Parameters:
+///   - lhsAddress: A pointer to (not a bitcast) a Swift type.
+///   - rhs: Another Swift type.
+///
+/// - Returns: Whether or not the type at `lhsAddress` and `rhs` are equivalent.
+///
+/// - Warning: This function is used to implement the `@Test` macro. Do not use
+///   it directly.
+public func __type(at lhsAddress: UnsafeRawPointer, is rhs: (some ~Copyable).Type) -> Bool {
+  let lhs = TypeInfo(describing: lhsAddress.load(as: (any ~Copyable.Type).self))
```

Comment:
This function is currently necessary because `==` is not defined for move-only metatypes (rdar://134276458), but once it is, we can just use it directly in the accessor function bodies.